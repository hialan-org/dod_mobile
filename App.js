import React from 'react';
import {Provider} from "react-redux";
import store from "./src/store";
import Navigation from "./src/pages/Navigation";
import * as Localization from 'expo-localization';
import i18n from 'i18n-js';

import {
    I18nManager,
} from "react-native";
import {translate} from "./src/i18n";

const translationGetters = {
    // lazy requires (metro bundler does not support symlinks)
    en: () => require("./src/i18n/en.json"),
};

const setI18nConfig = () => {
    // fallback if no available language fits
    const fallback = { locale: "en", isRTL: false };

    const { locale, isRTL } = Localization || fallback;
    console.log(locale);

    // clear translation cache
    translate.cache.clear();
    // update layout direction
    I18nManager.forceRTL(isRTL);
    // set i18n-js config
    i18n.translations = { [locale]: translationGetters[locale]() };
    i18n.locale = locale;
};

export default class App extends React.Component {

    constructor(props) {
        super(props);
        // setI18nConfig();
    }

    componentDidMount() {
    }

    render() {
        return (
            <Provider store={store}>
                <Navigation/>
            </Provider>
        );
    }
}