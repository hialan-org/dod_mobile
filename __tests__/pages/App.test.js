import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../App';

jest.mock('expo', () => {
    const Constants = {};
    Object.defineProperty(Constants, 'platform', { get: jest.fn() });
    return {
        SecureStore: {
            getItemAsync: jest.fn(),
            setItemAsync: jest.fn(),
            deleteItemAsync: jest.fn()
        },
        WebBrowser: {
            openAuthSessionAsync: jest.fn()
        },
        Constants
    };
});

describe.skip('<App />', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});