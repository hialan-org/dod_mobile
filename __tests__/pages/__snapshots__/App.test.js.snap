// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 1 child 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      onInsetsChange={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 64,
                }
              }
            >
              <View
                collapsable={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text>
                          Dog of The Dow
                        </Text>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "backgroundColor": "#DDDDDD",
                              "borderColor": "transparent",
                              "borderRadius": 2,
                              "borderStyle": "solid",
                              "borderWidth": 0,
                              "elevation": 0,
                              "height": 50,
                              "justifyContent": "center",
                              "marginTop": 16,
                              "minWidth": 64,
                              "padding": 10,
                              "width": 200,
                            }
                          }
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityStates={Array []}
                            accessible={true}
                            focusable={true}
                            isTVSelectable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "overflow": "hidden",
                                },
                                Object {
                                  "borderRadius": 2,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "marginLeft": 12,
                                    "marginRight": -4,
                                    "width": 16,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "uri": "https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Google_%22G%22_Logo.svg/200px-Google_%22G%22_Logo.svg.png",
                                    }
                                  }
                                  style={
                                    Object {
                                      "aspectRatio": 1,
                                      "width": 20,
                                    }
                                  }
                                />
                              </View>
                              <Text
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Array [
                                      Object {
                                        "letterSpacing": 1,
                                        "marginHorizontal": 16,
                                        "marginVertical": 9,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                      Object {
                                        "textTransform": "uppercase",
                                      },
                                      Object {
                                        "color": "#6200ee",
                                        "fontFamily": "System",
                                        "fontWeight": "500",
                                      },
                                      Object {
                                        "fontFamily": "System",
                                        "fontWeight": "500",
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  Sign-In with Google
                                </Text>
                              </Text>
                            </View>
                          </View>
                        </View>
                        <Text>
                          AppAuth: 
                          {
  "OAuthRedirect": [],
  "URLSchemes": []
}
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(224, 224, 224)",
                  "flex": 1,
                  "shadowColor": "rgb(224, 224, 224)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 64,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 20,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "marginHorizontal": 18,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;
